torch/_dynamo/variables/functions.py

```diff
@@ -319,61 +327,208 @@ def call_function(
+        return False
+
+    def __name__(self):
+        return self.get_name()
+
+    def __str__(self):
+        return f"{self.__class__.__name__}({self.get_name()})"
+
+    __repr__ = __str__
+
+    # def reconstruct(self, codegen):
+    #     from torch._dynamo.symbolic_convert import InstructionTranslator
+    #     tx = InstructionTranslator.current_tx()
+    #     try:
+    #         prev = tx.output.should_exit
+    #         tx.output.should_exit = False
+    #         items = self.force_unpack_var_sequence(tx)
+    #         variables.ListIteratorVariable(items, mutation_type=ValueMutationNew()).reconstruct(codegen)
+    #     finally:
+    #         tx.output.should_exit = prev
+
+    def bind_args(self, tx, args, kwargs):
+        return self.fn.bind_args(tx, args, kwargs)
+
+    def get_globals(self):
+        return self.f_globals
+
+    def python_type(self):
+        return types.GeneratorType
+
+    def _get_inline_tracer(self, tx):
+        from torch._dynamo.symbolic_convert import InliningInstructionTranslator
+
+        if self.inline_tracer is None:
+            self.inline_tracer = InliningInstructionTranslator.build_inline_tracer(
+                tx, self, [], {}
+            )
+        return self.inline_tracer
+
+    def next_variable(self, tx):
+        tracer = self._get_inline_tracer(tx)
+
+        try:
+            # Hierarchically, tx can be seen as the parent of the inline tracer
+            # created on call_function. Any exception needs to be propagated to tx
+            # for Dynamo to behave correctly
+            with patch.dict(counters, {"unimplemented": counters["inline_call"]}):
+                return tracer.inline_call_()
+        except ObservedException as e:
+            tx.exn_vt_stack.extend(tracer.exn_vt_stack)
+            raise e
+        except InfiniteGeneratorError:
+            # test/dynamo/test_misc.py::test_iterator_limit
+            raise
+        except Unsupported as e:
+            torch._C._dynamo.eval_frame.skip_code(self.get_code())
+            raise SkipFrame from e
+
+    def has_unpack_var_sequence(self, tx):
+        return False
+
+    def has_force_unpack_var_sequence(self, tx) -> builtins.bool:
+        return True
+
+    def force_unpack_var_sequence(self, tx) -> List[VariableTracker]:
+        result = []
+        while True:
+            try:
+                result.append(self.next_variable(tx))
+            except ObservedUserStopIteration:
+                handle_observed_exception(tx)
+                break
+        return result
 
+    def call_method(
+        self,
+        tx: "InstructionTranslator",
+        name: str,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        if name == "__next__":
+            return self.next_variable(tx)
+
+        super().call_method(tx, name, args, kwargs)
+
+
+class ContextlibContextManagerGeneratorObjectVariable(GeneratorObjectVariable):
+    """
```

Comment:
Inside comment, explain why this is special cased. E.g. our reconstruction strategy for generators is wrong but it doesn't matter most of the time (but it does matter in the context manager case)

Reply:
Also, would it be possible to have use GeneratorObjectVariable but instead have a flag saying if it's reconstructible or not?